VideoAPI/VideoApi.Services/ConsultationService.cs

```diff
@@ -0,0 +1,65 @@
+using System;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using VideoApi.Services.Contracts;
+using VideoApi.Services.Kinly;
+using Task = System.Threading.Tasks.Task;
+
+namespace VideoApi.Services
+{
+    public class ConsultationService : IConsultationService
+    {
+        private readonly IKinlyApiClient _kinlyApiClient;
+        private readonly ILogger<ConsultationService> _logger;
+
+        public ConsultationService(IKinlyApiClient kinlyApiClient, ILogger<ConsultationService> logger)
+        {
+            _kinlyApiClient = kinlyApiClient;
+            _logger = logger;
+        }
+        
+        public async Task<CreateConsultationRoomResponse> CreateConsultationRoomAsync(string virtualCourtRoomId, CreateConsultationRoomParams createConsultationRoomParams)
+        {
+            _logger.LogTrace("Creating a consultation for VirtualCourtRoomId: {virtualCourtRoomId} with prefix {createConsultationRoomParamsPrefix}",
+                virtualCourtRoomId, createConsultationRoomParams.Room_label_prefix);
+
+            CreateConsultationRoomResponse consultationRoomResponse;
+            try
+            {
+                consultationRoomResponse = await _kinlyApiClient.CreateConsultationRoomAsync(virtualCourtRoomId, createConsultationRoomParams);
+            }
+            catch (KinlyApiException e)
```

Comment:
i dont think this would work as expected,

_An Exception thrown within a Task is always wrapped in an AggregateException object. When using async/await with the await keyword, the first exception in the AggregateException is implicitly unwrapped. If more than one exception could be thrown by an execution path and all or some of them should be handled, it is recommended to keep a reference to the Task and access it’s Exception property._

Reply:
same below

Reply:
Removed the awaiter and returning the response immediately